### PR TITLE
Merchant Order Show page

### LINKED ISSUE #31: User Story 52, Merchant sees an order show page
```
As a merchant
When I visit an order show page from my dashboard
I see the customer's name and address
I only the items in the order that are being purchased from my inventory
I do not see any items in the order being purchased from other merchants
For each item, I see the following information:
- the name of the item, which is a link to my item's show page
- a small thumbnail of the item
- my price for the item
- the quantity the user wants to purchase
```

DO THESE STORIES FIRST:
- User Story 35, Merchant Dashboard displays Orders

### Checklist:

- [x] base tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #212

child of #274

depends on #231

### PR DESCRIPTION
**Pull request by BrennanAyers**
This PR brings in the functionality for a Merchant to view a Show page of one of their Orders. Since the Order might have items from multiple merchants, this Show page will only show relevant Items. 
The Show page shows the Ordering Users Name, and their full Address.
It also has a section for each Item belonging to the current Merchant that includes a link to an Item Show page, the thumbnail for the Item, the Price, and the Quantity that the User needs fulfilled.

Resolves #31

### CODE REVIEW

File: app/controllers/merchants/orders_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Merchants::OrdersController <ApplicationController
+  def show
+    @merchant = current_user
+    @order = Order.find(params[:id])
+    @user = @order.user
+    @order_items = @order.order_items.map {|order_item| order_item if order_item.item.user_id == @merchant.id}.compact
```
Comment (BrennanAyers) [inline app/controllers/merchants/orders_controller.rb:6]: This should probably change.

Comment (zachlp2016) [reply app/controllers/merchants/orders_controller.rb:6]: It works.

Comment (WHomer) [reply app/controllers/merchants/orders_controller.rb:6]: it's grate.

Comment (zachlp2016) [reply app/controllers/merchants/orders_controller.rb:6]: I'm not sure if this should be done in the model or not, but it certainly works.

**Review by WHomer — APPROVED** [review]
Swiss.

**Review by zachlp2016 — APPROVED** [review]
Testing is concise and directed.  It seems that sorting order_items by the merchants would be overkill in the model but may be what is required.  Though this looks good to me.
